apps/app/dashboards/national-accounts/iip/index.tsx

```diff
@@ -0,0 +1,306 @@
+    { label: t("keys.liabilities"), value: "liabilities" },
+  ];
+  const SHADE_OPTIONS: Array<OptionType> = [
+    { label: t("keys.no_shade"), value: "no_shade" },
+    { label: t("keys.recession"), value: "recession" },
+  ];
+
+  const { data, setData } = useData({
+    minmax: [0, timeseries.data[INDEX_OPTIONS[0].value].x.length - 1],
+    index_type: INDEX_OPTIONS[0],
+    shade_type: SHADE_OPTIONS[0],
+  });
+  const LATEST_TIMESTAMP =
+    timeseries.data[data.index_type.value].x[timeseries.data[data.index_type.value].x.length - 1];
+  const { coordinate } = useSlice(timeseries.data[data.index_type.value], data.minmax);
+
+  const formatToBillions = (number: number) => {
+    if (number >= 1e12) {
+      return `${numFormat(number / 1e9, "standard", 1, "long", i18n.language)} bil`;
+    }
+    return numFormat(number, "compact", [1, 1], "long", i18n.language, true);
+  };
+
+  const shader = useCallback<(key: string) => ChartDataset<keyof ChartTypeRegistry, any[]>>(
+    (key: string) => {
+      if (key === "no_shade")
+        return {
+          data: [],
+        };
+
+      return {
+        type: "line",
+        data: coordinate[key],
+        backgroundColor: AKSARA_COLOR.BLACK_H,
+        borderWidth: 0,
+        fill: true,
+        yAxisID: "y2",
+        stepped: true,
+      };
+    },
+    [data]
+  );
+
+  const getChartData = (sectionHeaders: string[]): TimeseriesChartData[] =>
+    sectionHeaders.map(chartName => ({
+      title: t(`keys.${chartName}`),
+      prefix: "RM ",
+      unitY: "",
+      label: t(`keys.${chartName}`),
+      data: coordinate[chartName],
+      fill: data.shade_type.value === "no_shade",
+      callout: {
+        latest:
+          timeseries_callout.data[data.index_type.value][chartName].latest > 0
+            ? `RM ${formatToBillions(
+                timeseries_callout.data[data.index_type.value][chartName].latest
+              )}`
+            : `-RM ${
+                formatToBillions(
+                  timeseries_callout.data[data.index_type.value][chartName].latest
+                ).split("-")[1]
+              }`,
+        change:
+          timeseries_callout.data[data.index_type.value][chartName].change > 0
+            ? `+RM ${formatToBillions(
+                timeseries_callout.data[data.index_type.value][chartName].change
+              )}`
+            : `-RM ${
+                formatToBillions(
+                  timeseries_callout.data[data.index_type.value][chartName].change
+                ).split("-")[1]
+              }`,
+      },
+      chartName,
+    }));
+
+  const detailsChartData = getChartData([
+    "derivatives",
+    "direct_debt",
+    "direct_equity",
+    "direct_total",
+    "other",
+    "porfolio_equity",
+    "portfolio_debt",
+    "portfolio_total",
+    "reserve",
+  ]);
+
+  useWatch(() => {
+    setData("minmax", [0, timeseries.data[data.index_type.value].x.length - 1]);
+  }, [data.index_type]);
```

Comment:
I think can remove this. Iinm, the reset is handled by the Slider component itself.